src/float_cast.h

```diff
@@ -106,17 +106,40 @@
 
 		return intgr ;
 	}
-#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_3X64)
+#elif (defined (WIN32) || defined (_WIN32)) && defined(_M_X64)
 
 	#include <math.h>
 	#include <immintrin.h>
+	#include <emmintrin.h>
 
-	__inline long int
-	lrintf (float flt)
+	#ifdef _MSC_VER
+		#pragma function(lrint, lrintf)
+	#endif
+
+	__inline 
+	long int lrint(double flt)
 	{
-		return _mm_cvt_ss2si(_mm_set_ss(flt));
+       return _mm_cvtsd_si32(_mm_set_sd(flt));
+    }
+
+	__inline 
+	long int lrintf (float flt)
+	{
+		return _mm_cvtss_si32(_mm_set_ss(flt));
```

Comment:
Converts to 32 bit integer, so assumes sizeof(long int) == 4

But is that right on 64 bit builds?

Surprise, yes:
https://docs.microsoft.com/en-us/cpp/cpp/data-type-ranges?view=msvc-160

I'd like to static_assert this surprising thing for the next reader!

And I'd like to review uses of lrint and lrintf in Audacity's own code, to be sure they are not wrong.

Because sometimes there are large times as doubles, that exceed 2^32 discrete samples -- you might record about 13.5 hours, not unheard of.  When quantizing to an integer, the type sampleCount is supposed to be used.

If data are lost in those conversions, the fault isn't in this function, it's in the calls to that function.